### PR TITLE
withdraw bug fix

### PR DESCRIPTION
**Pull request by 0xxlegolas**
Bug: the item index is reset when there is deposit on the same SSU for the second time. 

- Was fixed by setting the index to existing inventory itemList length if the SSU has some items

### CODE REVIEW

**Review by MerkleBoy — APPROVED** [review]
aside from the npm versioning error the code looks sound. LGTM 👍

File: end-to-end-tests/script/WithdrawFromInventory.s.sol [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    address invOwner = vm.addr(deployerPrivateKey);
+
+    // Start broadcasting transactions from the deployer account
+    vm.startBroadcast(deployerPrivateKey);
+    InventoryLib.World memory inventory = InventoryLib.World({
+      iface: IBaseWorld(worldAddress),
+      namespace: FRONTIER_WORLD_DEPLOYMENT_NAMESPACE
+    });
+
+    // uint256 smartObjectId = uint256(39041226111257929967735499841581212310485402913336957115444953092733099610999);
```
Comment (gunnigylfa) [inline end-to-end-tests/script/WithdrawFromInventory.s.sol:32]: Should this be commented out?

File: mud-contracts/world/package.json [hunk 2]
```diff
@@ -23,8 +23,8 @@
     "@latticexyz/world": "2.0.0-transaction-context-98ef570f",
     "@latticexyz/world-modules": "2.0.0-transaction-context-98ef570f",
     "@latticexyz/gas-report": "2.0.0-transaction-context-98ef570f",
-    "@eveworld/smart-object-framework": "0.0.1",
-    "@eveworld/common-constants": "0.0.1"
+    "@eveworld/smart-object-framework": "0.0.2",
```
Comment (gunnigylfa) [inline mud-contracts/world/package.json:26]: This is at a newer version

File: mud-contracts/world/package.json [hunk 3]
```diff
@@ -23,8 +23,8 @@
...
+    "@eveworld/common-constants": "0.0.2"
```
Comment (gunnigylfa) [inline mud-contracts/world/package.json:27]: This also

File: standard-contracts/package.json [hunk 4]
```diff
@@ -28,8 +28,8 @@
     "@latticexyz/schema-type": "2.0.0-transaction-context-98ef570f",
     "@latticexyz/world": "2.0.0-transaction-context-98ef570f",
     "@latticexyz/store": "2.0.0-transaction-context-98ef570f",
-    "@eveworld/world": "0.0.1",
-    "@eveworld/common-constants": "0.0.1"
+    "@eveworld/world": "0.0.2",
+    "@eveworld/common-constants": "0.0.2"
```
Comment (gunnigylfa) [inline standard-contracts/package.json:32]: This is at a newer version

File: end-to-end-tests/script/WithdrawFromInventory.s.sol (hunk 1 above)
Comment (0xxlegolas) [reply end-to-end-tests/script/WithdrawFromInventory.s.sol:32]: it was test id from game. will remove it

File: mud-contracts/world/package.json (hunk 2 above)
Comment (0xxlegolas) [reply mud-contracts/world/package.json:26]: this is the right version

File: mud-contracts/world/package.json (hunk 3 above)
Comment (0xxlegolas) [reply mud-contracts/world/package.json:27]: this is the right version

**Review by MerkleBoy — APPROVED** [review]
Bug is gone, congrats 👍

**Pull request merged**